### PR TITLE
Port rOpenScPCA package from OpenScPCA-analysis

### PR DESCRIPTION
**Pull request by sjspielman**
This PR adds the package at its current state, as well as the testing workflow. I also updated the license to just be BSD-3 since this code is now totally stand-alone.

### CODE REVIEW

File: .github/workflows/R-CMD-CHECK.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Set up R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: 4.4.0
```
Comment (jashapiro) [inline .github/workflows/R-CMD-CHECK.yml:21]: We may not want to peg this to R 4.4.0, but it should be fine for now.

**Review by jashapiro — APPROVED** [review]
LGTM

**Comment by jashapiro** [comment]
We will want to add the test workflow as required for merges to main and features, if you have not set that up yet. I can't see that part of the setting with the permissions I currently have.

**Comment by sjspielman** [comment]
> We will want to add the test workflow as required for merges to main and features,

Yup, that's on the way to being in place! Checks are required, and once this goes in I'll explicitly add the workflow here as a required status check.

**Pull request merged**